plugins/chat/app/models/chat/category_channel.rb

```diff
@@ -27,21 +27,5 @@ def generate_auto_slug
       self.slug = Slug.for(self.title.strip, "")
       self.slug = "" if duplicate_slug?
     end
-
-    def ensure_slug_ok
-      if self.slug.present?
-        # if we don't unescape it first we strip the % from the encoded version
-        slug = SiteSetting.slug_generation_method == "encoded" ? CGI.unescape(self.slug) : self.slug
-        self.slug = Slug.for(slug, "", method: :encoded)
-
-        if self.slug.blank?
-          errors.add(:slug, :invalid)
-        elsif SiteSetting.slug_generation_method == "ascii" && !CGI.unescape(self.slug).ascii_only?
-          errors.add(:slug, I18n.t("chat.category_channel.errors.slug_contains_non_ascii_chars"))
-        elsif duplicate_slug?
-          errors.add(:slug, I18n.t("chat.category_channel.errors.is_already_in_use"))
-        end
-      end
-    end
```

Comment:
Slug is not longer the specificity of category channels, every channel can now have a slug